packages/charts/react-charting/src/utilities/utilities.ts

```diff
@@ -320,6 +320,19 @@ export function createStringXAxis(
   return { xScale: xAxisScale, tickValues: dataset };
 }
 
+/**
+ * This method is used to calculate the rounded tick values for the y-axis
+ * @param {number} minVal
+ * @param {number} maxVal
+ * @param {number} splitInto
+ * @returns {number[]}
+ */
+function calculateRoundedTicks(minVal: number, maxVal: number, splitInto: number) {
+  const finalYmin = minVal >= 0 ? (minVal !== maxVal ? minVal : 0) : minVal;
```

Comment:
>finalYmin [](http://example.com/codeflow?start=8&length=9)

Check if this satisfies all cases.

const finalYmin = minVal >= 0 && minVal === maxVal ? 0 : minVal;
const finalYmax = minVal < 0 && minVal === maxVal ? 0 : maxVal;